### PR TITLE
Removed moment package, using Date and date-fns

### LINKED ISSUE #101: Remove moment dependency
We currently only use the `moment` library for two things:
1. calling `moment().toISOString()`
2. formatting the initial filename for SLP files

We do not need to import the entirety of `moment` just to use this functionality. We can simply use `(new Date()).toISOString()` and `import { format } from 'date-fns'` to format the date keeping in mind that the [format string is 🚨**different**🚨 to `moment`](https://date-fns.org/v2.28.0/docs/format).

### PR DESCRIPTION
**Pull request by shrianshChari**
Fixes #101
Tested with yarn, passed all test suites and tests

### CODE REVIEW

File: src/utils/slpFileWriter.ts [hunk 1]
```diff
@@ -8,18 +6,20 @@ import { SlpFile } from "./slpFile";
...
 import { SlpStream, SlpStreamEvent } from "./slpStream";
 
+import { format } from "date-fns";
+
 /**
  * The default function to use for generating new SLP files.
  */
-function getNewFilePath(folder: string, m: Moment): string {
-  return path.join(folder, `Game_${m.format("YYYYMMDD")}T${m.format("HHmmss")}.slp`);
+function getNewFilePath(folder: string, d: Date): string {
```
Comment (vinceau) [inline src/utils/slpFileWriter.ts:14]:
```suggestion
function getNewFilePath(folder: string, date: Date): string {
```
I know you just copied it from the old function but we should probably give this a proper variable name.

File: src/utils/slpFileWriter.ts [hunk 2]
```diff
@@ -8,18 +6,20 @@ import { SlpFile } from "./slpFile";
...
+  return path.join(folder, `Game_${format(d, "yyyymmdd")}T${format(d, "HHmmss")}.slp`);
```
Comment (vinceau) [inline src/utils/slpFileWriter.ts:15]:
```suggestion
  return path.join(folder, `Game_${format(d, "yyyyMMdd")}T${format(d, "HHmmss")}.slp`);
```
Make sure we don't confuse month and minutes.

**Review by vinceau — CHANGES_REQUESTED** [review]
Thanks for taking this one! Just some minor changes.

File: src/utils/slpFileWriter.ts (hunk 1 above)
Comment (shrianshChari) [reply src/utils/slpFileWriter.ts:14]: Yeah that probably makes sense, I'll finish refactoring on my machine

File: src/utils/slpFileWriter.ts (hunk 2 above)
Comment (shrianshChari) [reply src/utils/slpFileWriter.ts:15]: Right

**Review by vinceau — APPROVED** [review]
LGTM! Thanks for contributing!

**Pull request merged**